src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,36 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+	public final String value;
+    protected boolean isPrivate;
+    
+    
+    public Contact(){
+    	this.value = null;
+    	isPrivate = true;
+    }
+	
+	@Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Phone // instanceof handles nulls
+                && this.value.equals(((Phone) other).value)); // state check
```

Comment:
The equals method for the Contact class uses checks for the Phone class instead of its own class